MaxText/max_utils.py

```diff
@@ -274,4 +275,77 @@ def schedule(step):
     boundaries.append(warmup_steps + cos_steps + constant_zero_steps)
 
   return optax.join_schedules(pieces, boundaries)
-  
\ No newline at end of file
+
+
+@jax.custom_vjp
```

Comment:
Just make a note that this is from original t5x code as a ncomment. Looks great otherwise.

Reply:
Done.